ns-3_adapter/src/ns-3_adapter.cpp

```diff
@@ -66,9 +68,16 @@ void NS3Adapter::initialize() {
     pose_sub_ = pnh_->subscribe("current_pose", 1, &NS3Adapter::pose_cb, this);
 
     pnh_->getParam("vehicle_id", vehicle_id_);
+    pnh_->getParam("carla/ego_vehicle/role_name", role_id_);
+    pnh_->getParam("carla/port", port_);
```

Comment:
I think this should just be a parameter of the node, along with the host_ip below. I don't expect CARLA to provide these parameters.